snapcraft/internal/sources.py

```diff
@@ -389,7 +389,15 @@ def get(sourcedir, builddir, options):
     handler_class = _get_source_handler(source_type, options.source)
     handler = handler_class(options.source, sourcedir, source_tag,
                             source_branch)
-    handler.pull()
+    e_msg = 'A required package is missing, please install these packages: {}'
+    try:
+        handler.pull()
+    except FileNotFoundError:
+        if type(handler) in [Bazaar, Git, Subversion, Mercurial]:
+            required_packages = get_required_packages(options)
+            logging.error(e_msg.format(required_packages))
+        else:
```

Comment:
You might still need to raise an exception right? or else you break the logic of pull when those packages are not installed. You could raise a more specific exception in this case with that logging.error message instead of the logging.error call done here.